src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,287 @@
+                insertStmt.setLong(2, car.getId());
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't insert data of "
+                    + car + " into cars_drivers", exception);
+        }
+    }
+
+    private Car createCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("c_id", Long.class));
+        car.setModel(resultSet.getObject("c_m", String.class));
+        return car;
+    }
+
+    private Manufacturer createManufacturer(ResultSet resultSet) throws SQLException {
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setId(resultSet.getObject("m_id", Long.class));
+        manufacturer.setName(resultSet.getObject("m_n", String.class));
+        manufacturer.setCountry(resultSet.getObject("m_n", String.class));
+        return manufacturer;
+    }
+
+    private Driver creteDriver(ResultSet resultSet) throws SQLException {
+        Driver driver = new Driver();
+        driver.setId(resultSet.getObject("d_id", Long.class));
+        driver.setName(resultSet.getObject("d_n", String.class));
+        driver.setLicenceNumber(resultSet.getObject("d_ln", String.class));
+        return driver;
+    }
+
+    private Car getDriversByIdCars(Connection connection, Long id) {
+        String query = "select cars.id as c_id, cars.manufacturer_id as c_mid, "
+                + "cars.model as c_m, drivers.id as d_id, "
+                + "drivers.name as d_n, drivers.license_number as d_ln, "
+                + "manufacturers.id as m_id, manufacturers.name as m_n, "
+                + "manufacturers.country as m_c from cars "
+                + "inner join cars_drivers on cars_drivers.car_id = cars.id "
+                + "inner join drivers on drivers.id = cars_drivers.driver_id "
+                + "inner join manufacturers on manufacturers.id = cars.manufacturer_id "
+                + "where cars.id = ?;";
+        Car car = new Car();
+
+        try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                Manufacturer manufacturer = createManufacturer(resultSet);
+                Driver driver = creteDriver(resultSet);
+                car = createCar(resultSet);
+
+                car.setDrivers(driver);
+                car.setManufacturer(manufacturer);
+            }
+            while (resultSet.next()) {
+                Driver driver = creteDriver(resultSet);
+                car.setDrivers(driver);
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get drivers by car", exception);
+        }
+        return car;
+    }
+
+    private List<Car> getCarsByIdDrivers(Connection connection, Long id) {
```

Comment:
let's refactor the code so that we could remove this method as well

Reply:
done